commit e3ca33d10055d2e7609d1e72efa19bc4fb5832af
Author: Błażej Szczygieł <spaz16@wp.pl>
Date:   Fri Nov 22 23:48:03 2019 +0100

    macOS: Remove disabling iTunes on hotkey
    
    Doesn't work since 10.12.4 due to System Integrity Protection.

diff --git b/src/gui/Main.cpp a/src/gui/Main.cpp
index ea7436f4..3c32e09e 100644
--- b/src/gui/Main.cpp
+++ a/src/gui/Main.cpp
@@ -42,6 +42,9 @@
 #include <QBuffer>
 #include <QFile>
 #include <QDir>
+#ifdef Q_OS_MACOS
+    #include <QProcess>
+#endif
 #ifdef CHECK_FOR_EGL
     #include <QLibrary>
 #endif
@@ -51,6 +54,10 @@
 
 static ScreenSaver *g_screenSaver = nullptr;
 static bool g_useGui = true;
+#ifdef Q_OS_MACOS
+    static QByteArray g_rcdPath("/System/Library/LaunchAgents/com.apple.rcd.plist");
+    static bool g_rcdLoad;
+#endif
 
 /**/
 
@@ -336,6 +343,15 @@ static bool writeToSocket(IPCSocket &socket, QList<QPair<QString, QString>> &arg
 
 static inline void exitProcedure()
 {
+#ifdef Q_OS_MACOS
+    if (g_rcdLoad)
+    {
+        // Load RCD service again (allow to run iTunes on "Play" key)
+        QProcess::startDetached("launchctl load " + g_rcdPath);
+        g_rcdLoad = false;
+    }
+#endif
+
     delete g_screenSaver;
     g_screenSaver = nullptr;
 }
@@ -721,6 +737,16 @@ int main(int argc, char *argv[])
     HHOOK keyboardHook = SetWindowsHookEx(WH_KEYBOARD_LL, MMKeysHookProc, GetModuleHandle(nullptr), 0);
 #endif
 
+#ifdef Q_OS_MACOS
+    // Unload RCD service (prevent run iTunes on "Play" key)
+    {
+        QProcess launchctl;
+        launchctl.start("launchctl unload " + g_rcdPath);
+        if (launchctl.waitForFinished() && launchctl.exitStatus() == QProcess::NormalExit)
+            g_rcdLoad = !launchctl.readAllStandardError().startsWith(g_rcdPath);
+    }
+#endif
+
     qsrand(time(nullptr));
 
     do
